src/js/components/IndexPage.jsx

```diff
@@ -11,16 +11,21 @@ export class IndexPage extends PureComponent {
 
   // @todo remove forced condition
   // respond to search input
-  handleSubmit() {
-    return this.setState({ showResults: true });
+  handleSearchDone(data, searchTerm) {
+    if (!data.length) return this.setState({ showResults: false });
+    return this.setState({ showResults: true, resultData: data, searchTerm });
   }
 
   render(props) {
     return (
       <div>
-        <SearchBar handleSubmit={() => this.handleSubmit()} />
+        <SearchBar searchDone={(data, searchTerm) => this.handleSearchDone(data, searchTerm)} />
```

Comment:
Small thing, the convention in mobile for naming a function that occurs on some kind of event is to prefix with 'on'. So the equivalent in mobile (or rather, react-native-generic-table-page) is `onSearchChange`.

I find having 'Done' in the name kind of weird. Past tense. Reads like it expects a boolean.